src/main.rs

```diff
@@ -9,12 +9,73 @@
 
 #[macro_use] extern crate log;
 #[macro_use] extern crate clap;
+#[macro_use] extern crate error_chain;
 extern crate git2;
+extern crate libgitdit;
+
+mod error;
 
 use clap::App;
+use git2::Repository;
+use libgitdit::repository::RepositoryExt;
+use std::process::Command;
+
+use error::ErrorKind as EK;
+use error::*;
+
+
+/// Open the DIT repo
+///
+/// Opens the DIT repo corresponding to the current one honouring the user
+/// configuration.
+///
+fn open_dit_repo() -> Result<Repository> {
+    // TODO: access the config and maybe return another repo instead
+    Repository::open_from_env().chain_err(|| EK::WrappedGitError)
+}
+
+
+/// find-tree-init-hash subcommand implementation
+///
+fn find_tree_init_hash(repo: &Repository, matches: &clap::ArgMatches) -> Result<()> {
+    repo.revparse_single(matches.value_of("commit").unwrap())
+        .and_then(|obj| repo.find_commit(obj.id()))
+        .chain_err(|| EK::WrappedGitError)
+        .and_then(|commit| repo.find_tree_init(commit).chain_err(|| EK::WrappedGitDitError))
+        .map(|commit| println!("{}", commit.id()))
+}
+
+
+/// Handle unknown subcommands
+///
+/// Try to invoke an executable matching the name of the subcommand.
+///
+fn handle_unknown_subcommand(name: &str, matches: &clap::ArgMatches) -> Result<()> {
+    let mut command = Command::new(format!("git-dit-{}", name));
+    let result = try!(matches.values_of("")
+                             .unwrap()
```

Comment:
Currently, this will make the entire thing fail if the subcommand is invoked with zero arguments.